### PR TITLE
Adding nosetests for the podaac webservices

### PR DESCRIPTION
**Pull request by Omkar20895**
This Pull Requests addresses the tests for the podaac.py file. At present the following functions pass the test cases:

- load_image_granule
- list_available_granule_search_datasetIds
- list_available_granule_search_datasetShortNames
- list_available_granule_search_level2_datasetIds
- list_available_granule_search_level2_datasetShortNames
- list_available_image_granule_datasetIds
- list_available_image_granule_datasetShortNames
- list_available_extract_granule_datasetIds
- list_available_extract_granule_datasetShortNames

### CODE REVIEW

**Comment by lewismc** [comment]
@Omkar20895 all tests have failed
Please also remove all .DS_Store directories... these should not be included.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: Please add an Apache License v2.0 license header. You should be doing this for all of your source code files if you are coding at the Apache Software Foundation or related projects.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from .. import podaac
+from lxml import etree
+
+
+'''def test_load_dataset_md():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: Why is this commented out?

File: podaac/nose_tests_podaac/podaac_test.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
...CS', u'PODAAC-GH17L-2PE01', u'PODAAC-QSXXX-L1B02', u'PODAAC-MODAM-AN9N4', u'PODAAC-AQR40-3R1CS', u'PODAAC-QSBYU-ARILT', u'PODAAC-MODSA-MO4D4', u'PODAAC-MODSA-MO9N4', u'PODAAC-TELND-GO005', u'PODAAC-GHSEV-2PE01', u'PODAAC-AQR40-3R1AS', u'PODAAC-AQR40-3S7DS', u'PODAAC-GHCMC-4FM02', u'PODAAC-AQR40-3SCCS', u'PODAAC-GRGAD-20G05', u'PODAAC-MODTM-1D4N4', u'PODAAC-QSBYU-ANBIM', u'PODAAC-MODTM-MO4N4', u'PODAAC-MODSA-AN9N4', u'PODAAC-NSBYU-SNEN0', u'PODAAC-AQR40-3WCDS', u'PODAAC-TEOCN-EGC05', u'PODAAC-REYN2-OIMOW', u'PODAAC-TPMGD-BIN0B', u'PODAAC-AQR40-3DCCS', u'PODAAC-GHNOY-4FE01', u'PODAAC-REYN2-OIMON', u'PODAAC-GRGAD-20J15', u'PODAAC-GHTMI-2GR01', u'PODAAC-TELND-GIJP5', u'PODAAC-NSJPL-L3I02', u'PODAAC-AQR40-3ZMCS', u'PODAAC-AQR40-3WUCS', u'PODAAC-SEABY-ARBML', u'PODAAC-GRGSM-20GA5', u'PODAAC-PATHF-ANN51', u'PODAAC-NAVOC-2FRAC', u'PODAAC-GH18L-2PN01', u'PODAAC-SEABY-ARBMF', u'PODAAC-GRGAB-20G05', u'PODAAC-SPUR1-CTD00', u'PODAAC-AQR40-3W7CS', u'PODAAC-QSBYU-ARBML', u'PODAAC-SEABY-ANBMF', u'PODAAC-QSBYU-ARBMF', u'PODAAC-SEABY-ANBML', u'PODAAC-TEOCN-ETG05', u'PODAAC-PATHF-5DN51', u'PODAAC-PATHF-5DN50', u'PODAAC-GRGSM-96C05', u'PODAAC-TPTMR-REP01', u'PODAAC-AQR40-3SVCS', u'PODAAC-TEOCN-ENG05']
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The exact available_granule_search_datasetIds should not be hard coded. In it's current form this test can and will break based upon the dataset id's being added to or reduced. This is an unstable test... always.
I would suggest that you make it more general.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
...16_L', u'AQUARIUS_L3_WIND_SPEED_SMI_MONTHLY-CLIMATOLOGY_V4', u'AVHRR_PATHFINDER_L3_SST_8DAY_NIGHTTIME_V51', u'AVHRR_OI-NCEI-L4-GLOB-v2.0', u'TELLUS_OCEAN_NC_JPL_RL05', u'QUIKSCAT_BYU_L3_OW_SIGMA0_ENHANCED', u'EUR-L2P-AVHRR16_G', u'TELLUS_OCEANEOF_NC_JPL_RL05', u'AQUARIUS_L3_WIND_SPEED_SMID_7DAY_V4', u'AQUARIUS_L3_ANCILLARY_SST_SMI_DAILY_V4', u'AVHRR_PATHFINDER_L3_SST_5DAY_DAYTIME_V5', u'TELLUS_ECCO_OBP_KF080', u'WENTZ_SASS_SIGMA0_L2', u'AQUARIUS_L3_WIND_SPEED_SMID_28DAY-RUNNINGMEAN_V4', u'AQUARIUS_L3_WIND_SPEED_SMID_MONTHLY_V4', u'GRACE_GAD_L2_GRAV_GFZ_RL05', u'SEAWINDS_BYU_L3_OW_SIGMA0_ANTARCTIC_POLAR-STEREOGRAPHIC_BROWSE_MAPS_LTOD', u'SEAWINDS_BYU_L3_OW_SIGMA0_ARCTIC_POLAR-STEREOGRAPHIC_BROWSE_IMAGES_LTOD', u'TELLUS_OCEANEOF_TXT_GFZ_RL05', u'MODIS_AQUA_L3_SST_THERMAL_MONTHLY_4KM_NIGHTTIME_V2014.0', u'MTSAT2-OSPO-L2P-v1.0', u'AQUARIUS_L3_ANCILLARY_SST_SMI_3MONTH_V4', u'MODIS_T-JPL-L2P-v2014.0', u'UCLA_DEALIASED_SASS_L3', u'JASON-1_L2_OST_SGDR_VER-C_BINARY', u'GRACE_GAA_L2_GRAV_GFZ_RL05', u'TELLUS_LAND_GTIF_JPL_RL05', u'JPL-L4UHfnd-GLOB-MUR', u'AQUARIUS_L3_WIND_SPEED_SMIA_MONTHLY-CLIMATOLOGY_V4', u'TELLUS_LAND_NC_GFZ_RL05', u'AQUARIUS_L3_ANCILLARY_SST_SMIA_SEASONAL-CLIMATOLOGY_V4']
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this. It is not a good test and should not be hard coded.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	data = podaac.list_available_granule_search_datasetShortNames()
+
+	assert set(data) == set(test_data)
+
+def test_list_available_granule_search_level2_datasetIds():
+	test_data = ['PODAAC-AKASA-XOGD1', 'PODAAC-GHAM2-2PR72', 'PODAAC-ASOP2-25X01', 'PODAAC-ASOP2-12C01', 'PODAAC-ASOP2-25B01', 'PODAAC-ASOP2-COB01', 'PODAAC-GHAMT-2PE01', 'PODAAC-GHMDA-2PJ01', 'PODAAC-GHMDT-2PJ01', 'PODAAC-GH18G-2PN01', 'PODAAC-GH19G-2PN01', 'PODAAC-GHMTG-2PN01', 'PODAAC-OSCT2-L2BV2', 'PODAAC-J2ODR-GPS00', 'PODAAC-QSX12-L2B01', 'PODAAC-GHAMS-2PR01', 'PODAAC-GHVRS-2PN02']
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	data = podaac.list_available_granule_search_level2_datasetIds()
+
+	assert set(data) == set(test_data)
+
+def test_list_available_granule_search_level2_datasetShortNames():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+	test_data = ['ALTIKA_SARAL_L2_OST_XOGDR', 'AMSR2-REMSS-L2P-v7.2', 'ASCATA-L2-25km', 'ASCATA-L2-Coastal', 'ASCATB-L2-25km', 'ASCATB-L2-Coastal', 'EUR-L2P-AVHRR_METOP_A', 'JPL-L2P-MODIS_A', 'JPL-L2P-MODIS_T', 'NAVO-L2P-AVHRR18_G', 'NAVO-L2P-AVHRR19_G', 'NAVO-L2P-AVHRRMTA_G', 'OS2_OSCAT_LEVEL_2B_OWV_COMP_12_V2', 'OSTM_L2_OST_OGDR_GPS', 'QSCAT_LEVEL_2B_OWV_COMP_12', 'REMSS-L2P-AMSRE', 'VIIRS_NPP-NAVO-L2P-v1.0']
+	data = podaac.list_available_granule_search_level2_datasetShortNames()
+
+	assert set(data) == set(test_data)
+
+def test_list_available_image_granule_datasetIds():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+	assert set(data) == set(test_data)
+
+def test_list_available_image_granule_datasetIds(): 
+	test_data = ['PODAAC-AKASA-XOGD1', 'PODAAC-GHAM2-2PR72', 'PODAAC-ASOP2-25X01', 'PODAAC-ASOP2-12C01', 'PODAAC-ASOP2-25B01', 'PODAAC-ASOP2-COB01', 'PODAAC-GHAMT-2PE01', 'PODAAC-GHMDA-2PJ01', 'PODAAC-GHMDT-2PJ01', 'PODAAC-GH18G-2PN01', 'PODAAC-GH19G-2PN01', 'PODAAC-GHMTG-2PN01', 'PODAAC-OSCT2-L2BV2', 'PODAAC-J2ODR-GPS00', 'PODAAC-QSX12-L2B01', 'PODAAC-GHAMS-2PR01', 'PODAAC-GHVRS-2PN02']
+	data = podaac.list_available_image_granule_datasetIds()
+
+	assert set(data) == set(test_data)
+
+def test_list_available_image_granule_datasetShortNames():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+	test_data = ['ALTIKA_SARAL_L2_OST_XOGDR', 'AMSR2-REMSS-L2P-v7.2', 'ASCATA-L2-25km', 'ASCATA-L2-Coastal', 'ASCATB-L2-25km', 'ASCATB-L2-Coastal', 'EUR-L2P-AVHRR_METOP_A', 'JPL-L2P-MODIS_A', 'JPL-L2P-MODIS_T', 'NAVO-L2P-AVHRR18_G', 'NAVO-L2P-AVHRR19_G', 'NAVO-L2P-AVHRRMTA_G', 'OS2_OSCAT_LEVEL_2B_OWV_COMP_12_V2', 'OSTM_L2_OST_OGDR_GPS', 'QSCAT_LEVEL_2B_OWV_COMP_12', 'REMSS-L2P-AMSRE', 'VIIRS_NPP-NAVO-L2P-v1.0']
+	data = podaac.list_available_image_granule_datasetShortNames()
+
+	assert set(data) == set(test_data)	
+
+def test_list_available_extract_granule_datasetIds():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

File: podaac/nose_tests_podaac/podaac_test.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+	assert set(data) == set(test_data)	
+
+def test_list_available_extract_granule_datasetIds(): 
+	test_data = ['PODAAC-AKASA-XOGD1', 'PODAAC-GHAM2-2PR72', 'PODAAC-ASOP2-25X01', 'PODAAC-ASOP2-12C01', 'PODAAC-ASOP2-25B01', 'PODAAC-ASOP2-COB01', 'PODAAC-GHAMT-2PE01', 'PODAAC-GHMDA-2PJ01', 'PODAAC-GHMDT-2PJ01', 'PODAAC-GH18G-2PN01', 'PODAAC-GH19G-2PN01', 'PODAAC-GHMTG-2PN01', 'PODAAC-OSCT2-L2BV2', 'PODAAC-J2ODR-GPS00', 'PODAAC-QSX12-L2B01', 'PODAAC-GHAMS-2PR01', 'PODAAC-GHVRS-2PN02']
+	data = podaac.list_available_extract_granule_datasetIds()
+
+	assert set(data) == set(test_data)
+
+def test_list_available_extract_granule_datasetShortNames():
```
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: The same goes for this test.

**Comment by lewismc** [comment]
See why the tests are failing. It is due to a failed import 
https://travis-ci.org/lewismc/podaacpy/builds/132212570#L376

**Comment by Omkar20895** [comment]
@lewismc I get it now! This is because requests does not come by default with python, it needs to be installed using pip. My local machine had it installed already, so I overlooked it by mistake. I will fix it. Thanks

File: podaac/nose_tests_podaac/podaac_test.py (hunk 3 above)
Comment (Omkar20895) [inline podaac/nose_tests_podaac/podaac_test.py]: @lewismc If you look at the the function list_available_granule_search_datasetIds(), I have scraped the datasetIds from the website and returned it as a list. I realise that hardcoding is not a good practice but I couldn't find another elegant way of doing it. i would be glad if you could help me out here. :)

File: podaac/nose_tests_podaac/podaac_test.py (hunk 1 above)
Comment (Omkar20895) [inline podaac/nose_tests_podaac/podaac_test.py]: I will, Sorry.

File: podaac/nose_tests_podaac/podaac_test.py (hunk 3 above)
Comment (lewismc) [inline podaac/nose_tests_podaac/podaac_test.py]: Some initial tests you can perform
1) When making the call with requests, verify that the result is not null,
2) verify that there are more than one value in the response List
3) etc
The point is to create tests which are relevant for the long term as well as the short term

**Comment by Omkar20895** [comment]
Though I am installing requests the travis build is failing. I am creating a local image of travis using docker and will try solving the issue in a couple of hours. Thanks

**Comment by Omkar20895** [comment]
@lewismc I couldn't build an image of travis locally to test. Can I send some raw commits here to debug please?? Thanks

**Comment by lewismc** [comment]
You need to use the -H flag for installing requests so that it is available for the local user. Please read the yellow text in the image that you've provided.

**Comment by Omkar20895** [comment]
Yeah I read that, I talked to Ibrahim, he suggested me to remove sudo itself because it is not necessary here.

**Comment by Omkar20895** [comment]
Thanks @jarifibrahim for helping me out on fixing the build. @lewismc I am closing this pull request. I will be sending a new pull request with the right test cases and I am also done with the mcc.py file, I will send it soon.

**Comment by lewismc** [comment]
ok

**Pull request closed without merging**